site/src/sphinx/advanced-structured-logging.rst

```diff
@@ -148,6 +152,73 @@ automatically for you:
         }
     }
 
+Enabling content previews
+-------------------------
+Armeria provides the ``requestContentPreview`` and ``responseContentPreview`` properties in :api:`RequestLog`
+to retrieve the textual representation of the first N bytes of the request and response content.
+However, the properties are disabled by default so they always return ``null`` due to performance overhead.
+You can enable it when you configure :api:`Server`, :api:`VirtualHost` or :api:`Client`.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.server.ServerBuilder;
+    import com.linecorp.armeria.server.VirtualHostBuilder;
+
+    ServerBuilder sb = new ServerBuilder();
+    ...
+    // Enable previewing the content with the maxium length of 100 for textual content.
+    sb.contentPreview(100);
+    ...
+    VirtualHostBuilder vhb = new VirtualHostBuilder("http://example.com");
+    // In this case, the property of virtual host takes precedence over that of server.
+    vhb.contentPreview(150);
+    ...
+    sb.virtualHost(vhb.build());
+
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.ClientBuilder;
+    import com.linecorp.armeria.client.HttpClientBuilder;
+
+    ClientBuilder cb = new HttpClientBuilder();
+    ...
+    cb.contentPreview(100);
+
+Note that the properties are enabled only for textual contents and
+Armeria considers the following contents as textual contents.
+
+- when its type matches ``text/*`` or ``application/x-www-form-urlencoded``.
+- when its charset has been specified. e.g. application/json; charset=utf-8.
+- when its subtype is ``xml`` or ``json``. e.g. application/xml, application/json.
+- when its subtype ends with ``+xml`` or ``+json``. e.g. application/atom+xml, application/hal+json
+
+When a request or response begins, the :api:`ContentPreviewerFactory` set by a user
+creates a :api:`ContentPreviewer` based on HTTP headers to produce the preview when a request or response ends.
+
+You can configure :api:`Server`, :api:`VirtualHost`, or :api:`Client` to use your own :api:`ContentPreviewerFactory`
+and :api:`ContentPreviewer`. e.g.
+
+.. code-block:: java
+
+    import com.google.common.io.BaseEncoding;
+    import com.linecorp.armeria.common.logging.ContentPreviewer;
+
+    ServerBuilder sb = new ServerBuilder();
+
+    // A user can use their customized previewer factory.
+    sb.contentPreviewerFactory((ctx, headers) -> {
+        // the previewer which produces the preview through customized function
+        // when the contents have been aggregated more than specific bytes or the stream has been ended.
+        return ContentPreviewer.ofBinary(100, byteBuf -> {
+            // byteBuf has no more than 100 bytes.
+            byte[] contents = new byte[byteBuf.readableBytes()];
+            byteBuf.readBytes(contents);
+            return BaseEncoding.base16().encode(contents);
```

Comment:
Given that this incurs memory copy which may be expensive, how about using `ByteBufUtil.hexDump()`?